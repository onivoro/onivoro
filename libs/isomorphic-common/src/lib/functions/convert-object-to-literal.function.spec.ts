import { convertObjectToLiteral } from './convert-object-to-literal.function';

const qty = 337;
const price = 100;
const product = 'flux-capacitor';

describe(__filename, () => {
  it.each([[(k, v) => `${k}=${v}`, ' AND ', { qty, price, product }]])(
    'converts an object to a string',
    (literalFn, delimiter, obj) => {
      expect(
        convertObjectToLiteral(literalFn, delimiter, obj)
      ).toMatchSnapshot();
    }
  );
});
