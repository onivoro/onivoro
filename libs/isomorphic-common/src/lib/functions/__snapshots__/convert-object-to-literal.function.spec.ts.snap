// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/Users/lee/github.com/onivoro/onivoro/libs/isomorphic-common/src/lib/functions/convert-object-to-literal.function.spec.ts converts an object to a string 1`] = `"qty=337 AND price=100 AND product=flux-capacitor"`;
